packages/web-frontend/cypress/support/actions.js

```diff
@@ -4,14 +4,14 @@
  */
 
 import { TEST_USER } from '../constants';
-import { TestUserPasswordUndefinedError } from './helpers';
 
-const getTestUserPassword = () => {
-  const password = Cypress.env('TEST_USER_PASSWORD');
-  if (!password) {
-    throw new TestUserPasswordUndefinedError();
+const requireCyEnv = variable => {
+  const unprefixedVariable = variable.replace(/^CYPRESS_/, '');
```

Comment:
Slightly confused by this, so we are removing the `CYPRESS_` from the start of the variable? But later on we are calling it without a `CYPRESS_` prefix? Is this just to clean it? My final point of confusion is that aren't the variables prefixed with `CYPRESS_` in the `.env` file? Does Cyprus remove those prefixes when loading in variables?

Reply:
Cypress doesn't automatically load environment variables files from `.env` files (it uses its own config files)...We use [cypress-dotenv](https://www.npmjs.com/package/cypress-dotenv) for that.

`cypress-dotenv` only loads variables prefixed with `CYPRESS_` by default. Then, to actually retrieve the variables you have to omit the prefix... To be honest, I just realised there is a flag that you can pass to make it load all variables instead of just `CYPRESS_` ones.

I guess this is a standard practice when something is exposed to the client, probably in order to protect other sensitive data from being exposed, for example:
* [Gatsby env variables](https://www.gatsbyjs.com/docs/how-to/local-development/environment-variables/#client-side-javascript)
* [Create React app env variables](https://create-react-app.dev/docs/adding-custom-environment-variables/)

But this shouldn't be a concern here since the tests run on the server.

I still find it handy to prefix those variables with `CYPRESS_`, although I do understand that removing the prefix afterwards may seem a bit weird... Perhaps I should add a comment?

Reply:
Correction: it seems that the `CYPRESS_` prefix **can** (not **must**, as I thought) be ommitted. So both of the following work:

1. `Cypress.env(`CYPRESS_TEST_USER_PASSWORD`);
2.`Cypress.env(`TEST_USER_PASSWORD`);

The current implementation of the function supports both use cases, by removing the (optional) prefix before getting the value, and adding it when printing the name of the missing variable. It thus assumes that variables **must** be prefixed where they are actually defined